### PR TITLE
V2

### PR DESCRIPTION
**Pull request by kateloschinina**
Version of the header that we will be able to put into new components: groups and users

### CODE REVIEW

**Comment by kateloschinina** [comment]
What is done so far:
 - there is a footer that works in 'header' branch of groups

Outstanding issues:
 - not able to run locally standalone
 - mobile version is switched off
 - builds are failing

Please provide any comment on the current version of the code

File: server.js [hunk 1]
```diff
@@ -18,7 +21,8 @@ app.get('/', function (req, res) {
   res.render(__dirname + '/index', {'BASE_PATH_URL': process.env['BASE_PATH_URL']});
 });
 
-let port = process.env.PORT || 5000;
+// let port = process.env.PORT || 5000;
+let port = 5000;
```
Comment (dapenguin) [inline server.js]: Any reason why you are not allowing the port to be set using `process.env.PORT`?

File: src/reducers/license-change.js [hunk 2]
```diff
@@ -4,6 +4,9 @@ const defaultState = {
     const pathArr = urlPath.split('/');
     // the license should be the first item (not counting the 0 index which is an empty string)
     // replace it with the new id
+    if (pathArr[0] === '') {
+      pathArr.shift();
```
Comment (dapenguin) [inline src/reducers/license-change.js:8]: A little bit concerned that we are editing the array here, as `pathArr[1]` will now be pointing to a different array element to what it would have before this change.

What is the reason behind this change?

File: package-lock.json [hunk 3]
```diff
@@ -0,0 +1,18956 @@
+{
```
Comment (dapenguin) [inline package-lock.json]: Should this have been committed?

File: src/reducers/license-change.js (hunk 2 above)
Comment (kateloschinina) [reply src/reducers/license-change.js:8]: This is a newly created array, that just takes urlPath, and splits it by '/'; as it might start with '/' - sometimes we get empty string as the first element. With this check we remove it from array if that's the case.

File: server.js (hunk 1 above)
Comment (kateloschinina) [reply server.js]: There were conflicts with groups process.env.PORT, so the app didn't want to be run locally; reverted it back

File: Makefile [hunk 4]
```diff
@@ -5,7 +5,7 @@ node_modules/@financial-times/n-gage/index.mk:
 -include node_modules/@financial-times/n-gage/index.mk
 
 unit-test:
-	jest src/*
+	# skip
```
Comment (leggsimon) [inline Makefile]: do we have no tests for this?

File: circle.yml [hunk 5]
```diff
@@ -5,3 +5,7 @@ machine:
 dependencies:
   override:
     - npm prune && npm install && ./node_modules/.bin/bower install
+
+test:
```
Comment (leggsimon) [inline circle.yml:9]: we shouldn't need this command. our circleCI commands are configured for us, we ideally shouldn't override them unless we really have to

File: gulpfile.js [hunk 6]
```diff
@@ -10,24 +10,34 @@ let appServer;
 
 const verifyFn = function () {
   return obt.verify(gulp, {
-    scssLintPath: './.scss-lint.yml',
+    // scssLintPath: './.scss-lint.yml',
```
Comment (leggsimon) [inline gulpfile.js]: can this just be deleted?

File: gulpfile.js [hunk 7]
```diff
@@ -10,24 +10,34 @@ let appServer;
...
   return obt.build(gulp, {
-    js: './main.js',
+    js: './src/index.js',
     sass: './style/main.scss',
     buildJs: 'bundle.js',
-    buildCss: 'bundle.css',
+    buildCss: 'kmt-header.css',
     buildFolder: 'public',
-    scssLintPath: './.scss-lint.yml',
+    // scssLintPath: './.scss-lint.yml',
```
Comment (leggsimon) [inline gulpfile.js]: ditto

File: gulpfile.js [hunk 8]
```diff
@@ -10,24 +10,34 @@ let appServer;
...
     env: process.env.NODE_ENV
   });
 });
 
+gulp.task('build-page', function () {
+  return obt.build(gulp, {
+    sass: './style/page.scss',
+    buildCss: 'page.css',
+    buildFolder: 'public',
+    // scssLintPath: './.scss-lint.yml',
```
Comment (leggsimon) [inline gulpfile.js]: ditto

File: gulpfile.js [hunk 9]
```diff
@@ -84,5 +94,5 @@ gulp.task('dev-add-livereload', function () {
   process.env.DEV_ADD_LIVERELOAD = true;
 });
 
-gulp.task('default', ['build', 'img']);
+gulp.task('default', ['build', 'build-page', 'img']);
```
Comment (leggsimon) [inline gulpfile.js]: I don't think we want to `build-page` when we just run `gulp`. the `build-page` command just builds a demo right? It's fine to have a separate gulp task but I think that should only be called explicitly

File: package.json [hunk 10]
```diff
@@ -15,48 +15,58 @@
...
+    "chai-http": "^3.0.0",
+    "chai-jquery": "^2.0.0",
+    "enzyme": "^2.9.1",
+    "eslint-plugin-react": "^5.2.2",
+    "jest": "^20.0.0",
+    "jest-enzyme": "^3.1.0",
+    "jsdom": "^8.1.0",
+    "mocha": "^2.4.5",
+    "node-sass": "^3.8.0",
+    "origami-build-tools": "^5.5.3",
```
Comment (leggsimon) [inline package.json]: consider bumping this to v6. It brings in a newer version of node-sass which is quicker to install and will work with node 8 (can be done in a separate PR)

File: package.json [hunk 11]
```diff
@@ -15,48 +15,58 @@
...
-    "redux-thunk": "~2.1.0"
+    "babel-preset-stage-1": "^6.1.18",
+    "@financial-times/n-notification": "^6.0.0",
+    "dotenv": "^2.0.0",
+    "es6-promise": "^3.2.1",
+    "gulp": "^3.9.1",
+    "gulp-imagemin": "^2.2.1",
+    "gulp-livereload": "^3.8.1",
+    "gulp-nodemon": "^2.2.1",
+    "gulp-size": "^1.2.1",
```
Comment (leggsimon) [inline package.json]: All these gulp deps should be dev dependencies

File: src/actions/feedback-form.js [hunk 12]
```diff
@@ -26,12 +26,12 @@ export function toggleFeedbackValid () {
...
-export function submitFeedback (theUrl, data) {
+export function submitFeedback (data) {
   return (dispatch, getState) => {
-    const theStore = getState();
-    if (typeof theStore.KmtHeaderNs.helpers.doRequest === 'function') {
-      const options = {method: 'POST', body: JSON.stringify(data), headers: { 'Content-Type': 'application/json' }, 'mode': 'no-cors'};
-      theStore.KmtHeaderNs.helpers.doRequest(theUrl, options).then(() => {
+    // const theStore = getState();
+    // if (typeof theStore.KmtHeaderNs.helpers.doRequest === 'function') {
+      // const options = {method: 'POST', body: JSON.stringify(data), headers: { 'Content-Type': 'application/json' }, 'mode': 'no-cors'};
```
Comment (leggsimon) [inline src/actions/feedback-form.js]: Should this all be commented out? Can it be deleted?

File: src/helpers/helpers.js [hunk 13]
```diff
@@ -33,24 +33,27 @@ function _sortAndNameLicenses (licenses) {
  * Gets the user details from the cookie
  * @returns {{}}
  */
-export function getCookieUserDetails () {
+export function getCookieUserDetails (katConfig, licenceId) {
   // if the data has not been retrieved
-  if (licenceData === undefined) {
+  // if (licenceData === undefined) { - not sure why we needed it here
```
Comment (leggsimon) [inline src/helpers/helpers.js:38]: 😬 🤷‍♂️

File: src/index.js [hunk 14]
```diff
@@ -4,42 +4,53 @@ import { Provider } from 'react-redux';
...
+const store = configureStore();
+
+ReactDOM.render(
+	<Provider store={store}>
+		<KmtHeaderContainer />
+	</Provider>,
+	rootEl
+);
+//
+// class KmtHeader {
```
Comment (leggsimon) [inline src/index.js]: Can we remove all this?

**Review by leggsimon — APPROVED** [review]
LGTM! 👍 

Couple of things we could tidy up but nice one.

File: Makefile (hunk 4 above)
Comment (kateloschinina) [reply Makefile]: We do, and they fail right now, I guess because we changed react from v0.14 to v15. - there were some substantial changes in ReactTestUtils. Created separate card in Tech Hygiene for that.

File: gulpfile.js (hunk 6 above)
Comment (kateloschinina) [reply gulpfile.js]: ✅

File: gulpfile.js (hunk 7 above)
Comment (kateloschinina) [reply gulpfile.js]: ✅

File: gulpfile.js (hunk 8 above)
Comment (kateloschinina) [reply gulpfile.js]: ✅

File: circle.yml (hunk 5 above)
Comment (kateloschinina) [reply circle.yml:9]: If we delete it we get an issue with updating snapshots. https://circleci.com/gh/Financial-Times/kmt-header/223?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link - so definitely need this for now, but maybe issue need to be solved later on.... I don't know....

File: gulpfile.js (hunk 9 above)
Comment (kateloschinina) [reply gulpfile.js]: ... Makes sense I guess... ✅

File: src/index.js (hunk 14 above)
Comment (kateloschinina) [reply src/index.js]: Done! ✅

File: src/helpers/helpers.js (hunk 13 above)
Comment (kateloschinina) [reply src/helpers/helpers.js:38]: Let's keep it commented for now then, just in case

File: src/actions/feedback-form.js (hunk 12 above)
Comment (kateloschinina) [reply src/actions/feedback-form.js]: ✅

**Comment by kateloschinina** [comment]
There is another issue right now - I can't run the component standalone (without other apps, it was possible before, errors similar to what we were solving on Tuesday) - I guess this needs to be solved first, before creating a release?

File: package.json (hunk 11 above)
Comment (kateloschinina) [reply package.json]: https://trello.com/c/bboslv83/391-clean-dependencies-in-kmt-header

File: package.json (hunk 10 above)
Comment (kateloschinina) [reply package.json]: https://trello.com/c/bboslv83/391-clean-dependencies-in-kmt-header

**Review by leggsimon — APPROVED** [review]
